core/src/main/java/io/questdb/griffin/FunctionParser.java

```diff
@@ -252,8 +253,15 @@ public void visit(ExpressionNode node) throws SqlException {
             mutableArgPositions.clear();
             mutableArgPositions.setPos(argCount);
             for (int n = 0; n < argCount; n++) {
-                mutableArgs.setQuick(n, functionStack.poll());
-                mutableArgPositions.setQuick(n, positionStack.pop());
+                final Function arg = functionStack.poll();
+                final int pos = positionStack.pop();
+
+                if (arg instanceof GroupByFunction) {
+                    throw SqlException.position(pos).put("Aggregate function cannot be passed as an argument");
```

Comment:
`Function` is generally `closeable`. Some implementations allocate memory. When exception is thrown we need to ensure that functions are released.

Reply:
It's not very obvious what to clear at this point: `mutableArgs`, top of the `functionStack` or all items inside of it?
Is it safe to close Function several times?
Coz Function reference can be at different places.

Also, we have several throws places later in createFuntion:
https://github.com/questdb/questdb/blob/c4275e41560494cecdc350319d1aa8fde46bb16e/core/src/main/java/io/questdb/griffin/FunctionParser.java#L460
https://github.com/questdb/questdb/blob/c4275e41560494cecdc350319d1aa8fde46bb16e/core/src/main/java/io/questdb/griffin/FunctionParser.java#L646
https://github.com/questdb/questdb/blob/c4275e41560494cecdc350319d1aa8fde46bb16e/core/src/main/java/io/questdb/griffin/FunctionParser.java#L639
https://github.com/questdb/questdb/blob/c4275e41560494cecdc350319d1aa8fde46bb16e/core/src/main/java/io/questdb/griffin/FunctionParser.java#L320

Reply:
That is very cool way to present a comment, thank you 🎉 

As long as you're confident we're not going to leak memory out of allocating function, I'm happy to trust your judgement